mipengine/controller/api/algorithms_endpoint.py

```
@@ -50,45 +50,38 @@ async def get_algorithms() -> str:
+
+    # Validate the request
     try:
-        validate_algorithm_request(algorithm_name, request_body)
+        controller.validate_algorithm_execution_request(
+            algorithm_name=algorithm_name,
+            algorithm_request_dto=algorithm_request_dto)
     except (BadRequest, BadUserInput) as exc:
         raise exc
-    except:
-        logging.error(
-            f"Algorithm validation failed. Exception stack trace: \n {traceback.format_exc()}"
-        )
-        raise UnexpectedException()
-
-    try:
-        algorithm_request = AlgorithmRequestDTO.from_json(request_body)
-
-        # TODO: This looks freakin awful...
-        # Function run_algorithm_executor_in_threadpool calls the run method on the AlgorithmExecutor on a separate thread
-        # This function is queued in the running event loop
-        # Thus AlgorithmExecutor.run is executed asynchronoysly and does not block further requests to the server
-        def run_algorithm_executor_in_threadpool(algorithm_name, algorithm_request):
-            alg_ex = AlgorithmExecutor(algorithm_name, algorithm_request)
-
-            with concurrent.futures.ThreadPoolExecutor() as executor:
-                future = executor.submit(alg_ex.run)
-                result = future.result()
-                return result
-
-        loop = asyncio.get_running_loop()
-        algorithm_result = await loop.run_in_executor(
-            None,
-            run_algorithm_executor_in_threadpool,
-            algorithm_name,
-            algorithm_request,
-        )
-        return algorithm_result.json()
-
-    except:
-        logging.error(
-            f"Algorithm execution failed. Unexpected exception: \n {traceback.format_exc()}"
+    except: #TODO should not use bare except??
+        print(
+            f"Algorithm validation failed. Exception stack trace: \n"
+            f"{traceback.format_exc()}"
         )
         raise UnexpectedException()
+    
+    # Excute the requested Algorithm
```

As said before, what will happen if something breaks internally here? How will quart handle it if we don't throw an UnexpetedException that is currently handled as a 500?

---

you re right, I forgot to put that in a try/except. For now I just added a catch all block